### PR TITLE
Add a flag for including only certain fields

### PR DESCRIPTION
**Pull request by jalvz**
NOTE: this does not affect fields that are part of the title (timestamp, message, etc); I don't think it would be convenient for users to have to explicitly type those.

### CODE REVIEW

**Comment by trentm** [comment]
@jalvz Were you intending to start PRs for the three *issues*? Or did you want to go through them one at a time to minimize merge conflicts? Or if you don't have time to work through and finish PRs, that's totally cool tool.

File: cmd/ecslog/main_test.go [hunk 1]
```diff
@@ -96,9 +96,16 @@ var mainTestCases = []mainTestCase{
 		regexp.MustCompile(`^\n    foo: "bar"\n    spam: "eggs"\n$`),
 		nil,
 	},
+	{
+		"ecslog --include-fields ...all the title fields...",
+		[]string{"ecslog", "-i", "foo", "./testdata/exclude-fields.log"},
+		0,
+		regexp.MustCompile(`^\[2021-01-19T22:51:12.142Z\]  INFO: hi\n    foo: "bar"\n    spam: \n$`),
```
Comment (trentm) [inline cmd/ecslog/main_test.go:103]: Shouldn't the `spam: ...` line be complete gone here?

**Comment by trentm** [comment]
> NOTE: this does not affect fields that are part of the title (timestamp, message, etc); I don't think it would be convenient for users to have to explicitly type those.

I'm slightly hesitant to have "-i" fields not apply to the "title" line fields, mainly because the title line is underspecified. What is included in the title line by default may change in future versions and could also differ between formats.

However, I think I agree with you that it'll be useful as you have it.

(Aside: It also makes me want to have a "oneline" format that puts all the extra (non-title line) fields on the same line. Something like:

    $ cat some.log | ecslog -i foo,bar
    [2021-02-11T06:24:53.251Z]  WARN (myapi on purple.local): something went wrong
        foo: 1
        bar: 2
    ...
    $ cat some.log | ecslog -i foo,bar -f oneline
    [2021-02-11T06:24:53.251Z]  WARN (myapi on purple.local): something went wrong (foo:1 bar:2)
    ...

This is a case where I think `-i` and a more compact title line could really be nice.)


Anyway, that is a long way to say that I think your call to have `-i` not count for title line fields is a good call.

File: internal/ecslog/ecslog_test.go [hunk 2]
```diff
@@ -23,64 +24,88 @@ var renderFileTestCases = []renderFileTestCase{
...
 		"[2021-01-19T22:51:12.142Z]  INFO: hi\n",
 	},
 	{
 		"kql filtering, log.level range query, nope",
-		"no", "", "default", "", "log.level > warn",
+		"no", "", "default", "", "log.level > warn", []string{},
 		`{"log.level":"info","@timestamp":"2021-01-19T22:51:12.142Z","ecs":{"version":"1.5.0"},"message":"hi"}`,
 		"",
 	},
+	{
```
Comment (trentm) [inline internal/ecslog/ecslog_test.go:85]:
```suggestion

	// includeFields
	{
```

File: internal/ecslog/ecslog_test.go [hunk 3]
```diff
@@ -23,64 +24,88 @@ var renderFileTestCases = []renderFileTestCase{
...
+		"only log",
```
Comment (trentm) [inline internal/ecslog/ecslog_test.go:86]: Can you please prefix these test names with "includeFields: "?

File: internal/ecslog/ecslog_test.go [hunk 4]
```diff
@@ -23,64 +24,88 @@ var renderFileTestCases = []renderFileTestCase{
...
+		"no", "", "default", "", "", []string{"log"},
+		`{"log.level":"info","@timestamp":"2021-01-19T22:51:12.142Z","ecs":{"version":"1.5.0"},"message":"hi","log.origin":{"foo":"bar",file.name":"main.go","file.line":"42"}}`,
+		"{\"log.level\":\"info\",\"@timestamp\":\"2021-01-19T22:51:12.142Z\",\"ecs\":{\"version\":\"1.5.0\"},\"message\":\"hi\",\"log.origin\":{\"foo\":\"bar\",file.name\":\"main.go\",\"file.line\":\"42\"}}\n",
```
Comment (trentm) [inline internal/ecslog/ecslog_test.go:89]: Why doesn't this one render?

**Comment by jalvz** [comment]
My intention was to do one at a time as time permits, yes. Thanks for the review, I'll have a closer look shortly!

File: internal/ecslog/ecslog_test.go (hunk 4 above)
Comment (jalvz) [reply internal/ecslog/ecslog_test.go]: it should, was missing a `'`, fixed now

File: cmd/ecslog/main_test.go (hunk 1 above)
Comment (jalvz) [reply cmd/ecslog/main_test.go]: yes it, fixed

**Review by trentm — APPROVED** [review]

**Pull request merged**